include/sir/helpers.h

```diff
@@ -435,8 +439,8 @@ bool _sir_getchar(char* input);
  */
 # define _sir_snprintf_trunc(dst, size, ...) \
     do { \
-      volatile size_t _n = size; \
-      if (!snprintf(dst, _n, __VA_ARGS__)) { (void)_n; }; \
+      volatile size_t _n = (size_t)size; \
+      if (!snprintf(dst, (int)(size_t)_n, __VA_ARGS__)) { (void)_n; }; \
```

Comment:
`_n` is already `size_t`... why the cast?

Reply:
I'll see.

Reply:
Ah, now I see, with the override function:

```sh
src/sirerrors.c:211:17: error: implicit conversion loses integer precision: 'volatile size_t' (aka 'volatile unsigned long') to 'int' [-Werror,-Wshorten-64-to-32]
                _sir_snprintf_trunc(heap_msg, SIR_MAXERROR, _SIR_E_PLATFORM_ERRORFORMAT, _sir_te.os_error,
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/sir/helpers.h:439:26: note: expanded from macro '_sir_snprintf_trunc'
      if (!snprintf(dst, _n, __VA_ARGS__)) { (void)_n; }; \
           ~~~~~~~~      ^~
src/sirerrors.c:216:9: error: implicit conversion loses integer precision: 'volatile size_t' (aka 'volatile unsigned long') to 'int' [-Werror,-Wshorten-64-to-32]
        _sir_snprintf_trunc(message, SIR_MAXERROR, SIR_ERRORFORMAT, _sir_te.loc.func, //-V576
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/sir/helpers.h:439:26: note: expanded from macro '_sir_snprintf_trunc'
      if (!snprintf(dst, _n, __VA_ARGS__)) { (void)_n; }; \
           ~~~~~~~~      ^~
2 errors generated.
```

It's just getting tossed away, so the precision doesn't matter - the easiest way to always silence the warning in every case is just to cast it to short.  I'll do that.